app/src/main/java/com/example/learn09082021v001/DataInput.kt

```diff
@@ -3,68 +3,133 @@ package com.example.learn09082021v001
+
+class DataInput : AppCompatActivity(), AdapterUpdateRcv.OnclickRecycleView {
+    var StudentOOP = NameScoreOOP()
     override fun onSaveInstanceState(savedInstanceState: Bundle) {
         super.onSaveInstanceState(savedInstanceState)
         updateScoreOOP()
-        savedInstanceState.putSerializable(DATA_TRANFER, scoreOOP)
+        savedInstanceState.putSerializable(DATA_TRANFER, StudentOOP)
     }
 
     override fun onRestoreInstanceState(savedInstanceState: Bundle) {
         super.onRestoreInstanceState(savedInstanceState)
-        scoreOOP = savedInstanceState.getSerializable(DATA_TRANFER) as NameScoreOOP
+        StudentOOP = savedInstanceState.getSerializable(DATA_TRANFER) as NameScoreOOP
         upDateDataInput()
     }
 
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.exanble_manu, menu)
+        return super.onCreateOptionsMenu(menu)
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+        when (item.itemId) {
+            R.id.item1 -> Toast.makeText(this, "item 1 onclick", Toast.LENGTH_SHORT).show()
+        }
+        return super.onOptionsItemSelected(item)
+    }
+
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_data_input)
+        updateDataRecycleView()
         // BTN SET ONCLICK LISTENNER
         viewDataInputAndroid.getIncBtn().setOnClickListener {
-            scoreOOP.androiScoreINC()
-            viewDataInputAndroid.tvScore.text = scoreOOP.ScoreInPutandroid.toString()
+            StudentOOP.androiScoreINC()
+            viewDataInputAndroid.tvScore.text = StudentOOP.ScoreInPutandroid.toString()
         }
         viewDataInputAndroid.getDecBtn().setOnClickListener {
-            scoreOOP.androiScoreDEC()
-            viewDataInputAndroid.tvScore.text = scoreOOP.ScoreInPutandroid.toString()
+            StudentOOP.androiScoreDEC()
+            viewDataInputAndroid.tvScore.text = StudentOOP.ScoreInPutandroid.toString()
         }
         viewDataInputIos.getIncBtn().setOnClickListener {
-            scoreOOP.IOSScoreINC()
-            viewDataInputIos.tvScore.text = scoreOOP.ScoreInPutIOS.toString()
+            StudentOOP.IOSScoreINC()
+            viewDataInputIos.tvScore.text = StudentOOP.ScoreInPutIOS.toString()
         }
         viewDataInputIos.getDecBtn().setOnClickListener {
-            scoreOOP.IOSScoreDEC()
-            viewDataInputIos.tvScore.text = scoreOOP.ScoreInPutIOS.toString()
+            StudentOOP.IOSScoreDEC()
+            viewDataInputIos.tvScore.text = StudentOOP.ScoreInPutIOS.toString()
         }
-
         btnDone.setOnClickListener {
             updateScoreOOP()
             val intent = Intent(this, MainActivity::class.java)
-            intent.putExtra(DATA_TRANFER, scoreOOP)
+            intent.putExtra(DATA_TRANFER, StudentOOP)
+            updaterecycleView()
```

Comment:
the function `updaterecycleView` does init recyclerview rather than update. You should call it at `onCreate` one time only.
When need to update data to recyclerview, use `notifyDataSetChanged` instead https://developer.android.com/reference/androidx/recyclerview/widget/RecyclerView.Adapter#notifyDataSetChanged()